src/libServer/IsolatedServer.cpp

```diff
@@ -0,0 +1,291 @@
+                               ") lower than minimum for creating contract (" +
+                               to_string(CONTRACT_CREATE_GAS) + ")");
+  }
+
+  if (sender->GetNonce() >= tx.GetNonce()) {
+    throw JsonRpcException(ServerBase::RPC_INVALID_PARAMETER,
+                           "Nonce (" + to_string(tx.GetNonce()) +
+                               ") lower than current (" +
+                               to_string(sender->GetNonce()) + ")");
+  }
+
+  return true;
+}
+
+Json::Value IsolatedServer::CreateTransaction(const Json::Value& _json) {
+  try {
+    if (!JSONConversion::checkJsonTx(_json)) {
+      throw JsonRpcException(RPC_PARSE_ERROR, "Invalid Transaction JSON");
+    }
+
+    LOG_GENERAL(INFO, "On the isolated server ");
+
+    Transaction tx = JSONConversion::convertJsontoTx(_json);
+
+    Json::Value ret;
+
+    const Address fromAddr = tx.GetSenderAddr();
+    const Account* sender = AccountStore::GetInstance().GetAccount(fromAddr);
+
+    if (!ValidateTxn(tx, fromAddr, sender, m_gasPrice)) {
+      return ret;
+    }
+
+    if (sender->GetNonce() + 1 != tx.GetNonce()) {
+      throw JsonRpcException(
+          RPC_INVALID_PARAMETER,
+          "Expected Nonce: " + to_string(sender->GetNonce() + 1));
+    }
+
+    if (sender->GetBalance() < tx.GetAmount()) {
+      throw JsonRpcException(
+          RPC_INVALID_PARAMETER,
+          "Insufficient Balance: " + sender->GetBalance().str());
+    }
+
+    if (m_gasPrice > tx.GetGasPrice()) {
+      throw JsonRpcException(RPC_INVALID_PARAMETER,
+                             "Minimum gas price greater: " + m_gasPrice.str());
+    }
+
+    switch (Transaction::GetTransactionType(tx)) {
+      case Transaction::ContractType::NON_CONTRACT:
+        break;
+      case Transaction::ContractType::CONTRACT_CREATION:
+        if (!ENABLE_SC) {
+          throw JsonRpcException(RPC_MISC_ERROR, "Smart contract is disabled");
+        }
+        ret["ContractAddress"] =
+            Account::GetAddressForContract(fromAddr, sender->GetNonce()).hex();
+        break;
+      case Transaction::ContractType::CONTRACT_CALL: {
+        if (!ENABLE_SC) {
+          throw JsonRpcException(RPC_MISC_ERROR, "Smart contract is disabled");
+        }
+        const Account* account =
+            AccountStore::GetInstance().GetAccount(tx.GetToAddr());
+
+        if (account == nullptr) {
+          throw JsonRpcException(RPC_INVALID_ADDRESS_OR_KEY, "To addr is null");
+        }
+
+        else if (!account->isContract()) {
+          throw JsonRpcException(RPC_INVALID_ADDRESS_OR_KEY,
+                                 "Non - contract address called");
+        }
+      } break;
+
+      case Transaction::ContractType::ERROR:
+        throw JsonRpcException(RPC_INVALID_ADDRESS_OR_KEY,
+                               "Code is empty and To addr is null");
+        break;
+      default:
+        throw JsonRpcException(RPC_MISC_ERROR, "Txn type unexpected");
+    }
```

Comment:
Seems all these checks are duplicate from Lookup Server, since the purpose of this isolated server is to immidate the behavior of txn, for consistency and the ease of maintainance. It's better to encapsulate these checks in LookupServer class into a separte public function and refer it here.

Reply:
The lookup server function is a static one, so there is no easy way to achieve this.